### PR TITLE
add option to override default levels

### PR DESCRIPTION
**Pull request by Shareed2k**

### CODE REVIEW

File: airbrake.go [hunk 1]
```diff
@@ -10,13 +10,21 @@ import (
...
+	logrus.ErrorLevel,
+	logrus.FatalLevel,
+	logrus.PanicLevel,
+}
+
 // AirbrakeHook to send exceptions to an exception-tracking service compatible
 // with the Airbrake API.
 type airbrakeHook struct {
 	Airbrake *gobrake.Notifier
+	levels   []logrus.Level
```
Comment (gravis) [inline airbrake.go]: Why not making this field exported instead?

File: airbrake.go [hunk 2]
```diff
@@ -10,13 +10,21 @@ import (
...
 }
 
-func NewHook(projectID int64, apiKey, env string) *airbrakeHook {
+func NewHook(projectID int64, apiKey, env string, levels ...logrus.Level) *airbrakeHook {
```
Comment (gravis) [inline airbrake.go]: If `levels` was exported (see my comment above), we could keep the same API.
Users would have to use `hook.Levels = []logrus.Level{logrus.ErrorLevel,
		logrus.FatalLevel}` for example.

**Review by gravis — COMMENTED** [review]
Added comments about leaving the current API as it, and use an exported field instead.

File: airbrake.go (hunk 2 above)
Comment (Shareed2k) [reply airbrake.go]: no problem to export the field Levels  but need only change Levels field name cause it has both field and method named 'Levels'.

**Comment by Shareed2k** [comment]
can you please create a new release

вт, 25 февр. 2020 г. в 02:44, Philippe Lafoucrière <notifications@github.com

**Comment by gravis** [comment]
Done! [4.1.0 is available](https://github.com/gemnasium/logrus-airbrake-hook/releases/tag/v4.1.0).

**Comment by Shareed2k** [comment]
thank you!

вт, 25 февр. 2020 г. в 14:49, Philippe Lafoucrière <notifications@github.com

**Pull request merged**